mountpoint-s3-crt/src/s3/client.rs

```diff
@@ -1181,14 +1183,19 @@ impl RequestMetrics {
 
     /// Get the ID of the thread the request was made from
     pub fn thread_id(&self) -> Option<ThreadId> {
-        let mut out: aws_thread_id_t = 0;
-        // SAFETY: `inner` is a valid aws_s3_request_metrics
-        unsafe {
-            aws_s3_request_metrics_get_thread_id(self.inner.as_ptr(), &mut out)
+        let mut out: MaybeUninit<aws_thread_id_t> = MaybeUninit::uninit();
+        // SAFETY: `inner` is a valid aws_s3_request_metrics, `out` will be a valid pointer if no error was set
+        let thread_id = unsafe {
+            aws_s3_request_metrics_get_thread_id(self.inner.as_ptr(), out.as_mut_ptr())
                 .ok_or_last_error()
-                .ok()?
+                .ok()?;
+            assert!(
+                !out.as_ptr().is_null(),
+                "Thread ID should be available if call succeeded",
+            );
```

Comment:
This assertion is always true, `out.as_ptr()` is a pointer _to the_ `aws_thread_id_t`. I think you can just drop it; the early return tells us that it succeeded.

Reply:
Got it. In this case, we're giving the function a pointer to a memory location to populate and we then assume its initialized after the call completes successfully.

We'd want to check the pointer isn't null if we passed a _null pointer_ where the function should replace that with a valid pointer and wanted to check its been updated to a new location of memory.